### PR TITLE
Vb 2361 add session template active true false

### PR DESCRIPTION
**Pull request by dhirajshettymoj**
## What does this pull request do?

Adds the active flag to activate / deactivate session templates.
## What is the intent behind these changes?

VB-2361 implementation.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt [hunk 1]
```diff
@@ -28,13 +28,16 @@ data class UpdateSessionTemplateDto(
...
 
-  @Schema(description = "list of group references for permitted session location groups", required = false)
-  val locationGroupReferences: List<String>? = null,
-
   @Schema(description = "number of weeks until the weekly day is repeated", example = "1", required = true)
   @field:Min(1)
   val weeklyFrequency: Int?,
 
+  @Schema(description = "is prison active", example = "true", required = false)
+  val active: Boolean? = null,
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt:36]: do we need this here, maybe?

Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt:36]: do we need this here, maybe?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt [hunk 2]
```diff
@@ -51,7 +52,7 @@ interface SessionTemplateRepository : JpaRepository<SessionTemplate, Long> {
       "where u.prison.code = :prisonCode " +
```
Comment (AledWEvans) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt:52]: `findActiveAndFutureSessionTemplates`
should be
`findCurrentAndFutureSessionTemplates`

**Review by AledWEvans — COMMENTED** [review]
Some comments have a read

File: src/main/resources/db/migration/V2_28__add_active_flag_session_template.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+ALTER TABLE session_template ADD active bool NOT NULL DEFAULT true;
```
[the remark this answers is not in the document]
Comment (dhirajshettymoj) [reply src/main/resources/db/migration/V2_28__add_active_flag_session_template.sql]: Set default to false and added script to set existing session templates to true.

[the remark this answers is not in the document]
Comment (AledWEvans) [reply src/main/resources/db/migration/V2_28__add_active_flag_session_template.sql]: good call

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt (hunk 1 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt]: Removed from both create and update

Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/dto/sessions/UpdateSessionTemplateDto.kt]: True, not needed. Removed from both create and update as suggested.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt (hunk 2 above)
Comment (dhirajshettymoj) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/repository/SessionTemplateRepository.kt:52]: Updated

**Review by AledWEvans — APPROVED** [review]
LGTM

**Pull request merged**